core-bundle/src/EventListener/DataContainer/FormOverwriteListener.php

```diff
@@ -33,35 +41,12 @@ public function addOverwritableFields(string $table): void
         System::loadLanguageFile('tl_form');
```

Comment:
Is there any way load the language files via a service? I know I could use `\Contao\DcaLoader` for the data container, but I did not find a suitable replacement for `System::loadLanguageFile`

Reply:
If you need the `$GLOBALS` array to be populated that's the only way to do AFAIK. For everything else you can use the translator service.

Reply:
The translator service would also populate the `$GLOBALS` array if you use something like `->trans(…, 'contao_tl_form')` but I’m not sure if you should rely on that.

An alternative would be to inject the framework and use `$this->framework->getAdapter(System::class)->loadLanguageFile('tl_form');`

Reply:
Thanks for your hints, I'll use the adapter pattern for the language files then 👍